### PR TITLE
Add blobstorage service

### PR DESCRIPTION
**Pull request by maxXximus23**
- Add azure blob storage package;
- Set up appsettings and Startup for azure blob storage;
- Add Blob storage Service.

### CODE REVIEW

**Comment by kirpichyov** [comment]
Please verify your changes...

File: AchieveIt.API/Controllers/AuthController.cs [hunk 1]
```diff
@@ -12,13 +11,15 @@ namespace AchieveIt.API.Controllers
     [Authorize(Roles = "Admin")]
     public class AuthController : ControllerBase
     {
+        private readonly IBlobService _blobService;
         private readonly IAuthService _authService;
         private readonly IMapper _mapper;
 
-        public AuthController(IAuthService authService, IMapper mapper)
+        public AuthController(IAuthService authService, IMapper mapper, IBlobService blobService)
```
Comment (kirpichyov) [inline AchieveIt.API/Controllers/AuthController.cs]: Please remove. Looks like added unintentionally.

File: AchieveIt.BusinessLogic/Services/BlobService.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        public async Task DeleteBlobAsync(string blobName)
+        {
+            var containerClient = _blobServiceClient.GetBlobContainerClient("avatars");
+            var blobClient = await containerClient.GetBlobClient(blobName).DeleteAsync();
+        }
+
+        public Task<IEnumerable<string>> ListBlobAsync()
+        {
+            throw new System.NotImplementedException();
```
Comment (kirpichyov) [inline AchieveIt.BusinessLogic/Services/BlobService.cs]: Please remove this method (from the contract as well).

File: AchieveIt.API/Program.cs [hunk 3]
```diff
@@ -11,7 +11,6 @@ public class Program
         public static void Main(string[] args)
         {
             var configuration = new ConfigurationBuilder()
-                .AddJsonFile("appsettings.json")
```
Comment (kirpichyov) [inline AchieveIt.API/Program.cs]: Looks like that will lead to the problems with logging

File: AchieveIt.BusinessLogic/Services/BlobService.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        private readonly BlobStorageOptions _blobStorageOptions;
+        public BlobService(BlobServiceClient blobServiceClient, IOptions<BlobStorageOptions> blobStorageOptions)
+        {
+            _blobServiceClient = blobServiceClient;
+            _blobStorageOptions = blobStorageOptions.Value;
+        }
+
+        public async Task<BlobFileInfo> GetBlobAsync(string name)
+        {
+            var containerClient = _blobServiceClient.GetBlobContainerClient("avatars");
```
Comment (kirpichyov) [inline AchieveIt.BusinessLogic/Services/BlobService.cs]: Looks like should be moved to parameter

File: AchieveIt.BusinessLogic/Services/BlobService.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            var blobClient = containerClient.GetBlobClient(name);
+            var blobDownloadInfo = await blobClient.DownloadAsync();
+
+            return new BlobFileInfo(blobDownloadInfo.Value.Content, blobDownloadInfo.Value.ContentType);
+        }
+
+        public async Task UploadFileBlobAsync(IFormFile file, bool inline = false)
+        {
+            var container = _blobServiceClient.GetBlobContainerClient("avatars");
```
Comment (kirpichyov) [inline AchieveIt.BusinessLogic/Services/BlobService.cs]: Looks like should be moved to parameter

File: AchieveIt.BusinessLogic/Services/BlobService.cs [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            var blob = container.GetBlobClient(Guid.NewGuid().ToString());
+            var header = new BlobHttpHeaders()
+            {
+                ContentType = file.ContentType,
+                ContentDisposition = $"{(inline? "inline" : "attachment")}; filename=\"{file.FileName}\";"
```
Comment (kirpichyov) [inline AchieveIt.BusinessLogic/Services/BlobService.cs]: Suggestion for formatting:
`inline ?`

**Review by kirpichyov — CHANGES_REQUESTED** [review]

File: AchieveIt.API/Controllers/AuthController.cs (hunk 1 above)
Comment (maxXximus23) [reply AchieveIt.API/Controllers/AuthController.cs]: Done.

File: AchieveIt.BusinessLogic/Services/BlobService.cs (hunk 2 above)
Comment (maxXximus23) [reply AchieveIt.BusinessLogic/Services/BlobService.cs]: Done.

File: AchieveIt.API/Program.cs (hunk 3 above)
Comment (maxXximus23) [reply AchieveIt.API/Program.cs]: Done.

File: AchieveIt.BusinessLogic/Services/BlobService.cs (hunk 4 above)
Comment (maxXximus23) [reply AchieveIt.BusinessLogic/Services/BlobService.cs]: Done.

File: AchieveIt.BusinessLogic/Services/BlobService.cs (hunk 5 above)
Comment (maxXximus23) [reply AchieveIt.BusinessLogic/Services/BlobService.cs]: Done.

File: AchieveIt.BusinessLogic/Services/BlobService.cs (hunk 6 above)
Comment (maxXximus23) [reply AchieveIt.BusinessLogic/Services/BlobService.cs]: Done.

**Review by kirpichyov — APPROVED** [review]

**Review by qrilo — APPROVED** [review]

**Pull request merged**